syntax = "proto3";


service Hashtags {
    rpc sendTweet (TweetRequest) returns (Empty) {}
    rpc getTweetsByHashtag (TweetHashtagRequest) returns (TweetsList) {}
    rpc getTweetSentiment (TweetHashtagRequest) returns (TweetSentiment) {}
}

message Empty {}

message TweetRequest {
    string tweet = 1;
    int32 tweet_id = 2;
}

message TweetHashtagRequest {
    string hashtag = 1;
}

message TweetsList {
    string hashtag = 1;
    repeated string tweets = 2;
}

message TweetSentiment {
    enum Sentiments {
        POSITIVE = 0;
        NEGATIVE = 1;
        NEUTRAL = 2;
    }

    Sentiments sentiment = 1;
}
